src/mqueryfront/src/QueryMonaco.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from "react";
+import Editor, { monaco } from "@monaco-editor/react";
+import YARA from "./yara-lang";
+
+class QueryMonaco extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            rawYara: props.rawYara,
+            editor: null,
+            readOnly: false,
+            decorations: [],
+        };
+        this.handleEditorDidMount = this.handleEditorDidMount.bind(this);
+    }
+
+    getValue() {
+        if (this.state.editor) {
+            return this.state.editor.getValue();
+        }
+    }
+
+    setValue(newValue) {
+        if (this.state.editor) {
+            this.state.editor.setValue(newValue);
+        }
+    }
+
+    setReadOnly(readOnly) {
+        if (this.state.editor) {
+            this.setState({ readOnly: readOnly });
+        }
+    }
+
+    setError(error) {
+        if (!this.state.editor || !error) {
+            return;
+        }
+
+        monaco.init().then((monaco) => {
+            this.state.decorations = this.state.editor
+                .getModel()
+                .deltaDecorations(
+                    [],
+                    [
+                        {
+                            range: new monaco.Range(
+                                error[1],
+                                error[2],
+                                error[1],
+                                Number(error[3]) + 1 // Increase by one because it excludes end column
+                            ),
+                            options: {
+                                isWholeLine: error.length > 3 ? false : true,
```

Comment:
According to the previous suggestion:
```suggestion
                                isWholeLine: endColumn !== undefined ? false : true,
```

Reply:
omg I have no idea why I used if ternary here instead of just 
```
isWholeLine: error.length != 4
```
or in the new suggestion -> 
```
isWholeLine: encColumn == undefined
```